src/lightning/pytorch/callbacks/gradient_accumulation_scheduler.py

```diff
@@ -121,17 +122,21 @@ def on_train_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule")
 
         # local import to avoid circular import
         from lightning.pytorch.accelerators import IPUAccelerator
-        from lightning.pytorch.strategies import ColossalAIStrategy, DeepSpeedStrategy
+        from lightning.pytorch.strategies import DeepSpeedStrategy
 
-        unsupported_strategies = (DeepSpeedStrategy, ColossalAIStrategy)
         unsupported_accelerators = (IPUAccelerator,)
+        unsupported_strategies = [DeepSpeedStrategy]
+        if _LIGHTNING_COLOSSALAI_AVAILABLE:
+            from lightning_colossalai import ColossalAIStrategy
+
+            unsupported_strategies.append(ColossalAIStrategy)
```

Comment:
We decoupled the callback from the trainer/strategy in #16729. I would prefer to keep it that way so that it doesn't matter where the callback lives.